### PR TITLE
testing/boost-build: upgrade to 1.66.0

### PR DESCRIPTION
**Pull request by vakartel**

### CODE REVIEW

File: testing/boost-build/APKBUILD [hunk 1]
```diff
@@ -1,38 +1,35 @@
...
 url="http://boost.org/build"
 arch="all"
+options="!check"
 license="custom:boost"
 depends="boost-dev"
 makedepends=""
 subpackages="$pkgname-doc"
 source="$pkgname-$pkgver.tar.gz::https://github.com/boostorg/${pkgname#*-}/archive/${pkgname%%-*}-$pkgver.tar.gz
-	$pkgname-$pkgver-LICENSE.txt::http://boost.org/LICENSE_1_0.txt"
+	$pkgname-LICENSE.txt::http://boost.org/LICENSE_1_0.txt"
```
Comment (jirutka) [inline testing/boost-build/APKBUILD]: Why did you change this?

Comment (vakartel) [reply testing/boost-build/APKBUILD]: Because license file not assigned to package version and not changes between versions.

**Comment by andypost** [comment]
Related https://github.com/alpinelinux/aports/pull/2960

File: testing/boost-build/APKBUILD (hunk 1 above)
Comment (jirutka) [reply testing/boost-build/APKBUILD]: That does not matter. What matters is that source names must be unique among all aports in all versions. If Boost change this file or release new version and you keep the current name, it will fail on builders because sources are cached and identified only by name.
